src/nvim/regexp_nfa.c

```diff
@@ -1290,32 +1290,30 @@ static int nfa_regatom(void)
     break;
 
   case Magic('('):
-    if (nfa_reg(REG_PAREN) == FAIL)
-      return FAIL;                  /* cascaded error */
+    if (nfa_reg(REG_PAREN) == FAIL) {
+        return FAIL;                  // cascaded error
+    }
     break;
 
   case Magic('|'):
   case Magic('&'):
   case Magic(')'):
-    EMSGN(_(e_misplaced), no_Magic(c));
-    return FAIL;
-
   case Magic('='):
   case Magic('?'):
   case Magic('+'):
   case Magic('@'):
   case Magic('*'):
   case Magic('{'):
-    /* these should follow an atom, not form an atom */
+    // these should follow an atom, not form an atom */
```

Comment:
this comment does not apply to `Magic(')')` and the other conflated cases, so it is confusing to conflate the cases. 

Better to suppress the PVS warning.